hal/inc/socket_hal.h

```diff
@@ -92,7 +92,21 @@ sock_result_t socket_accept(sock_handle_t sd);
  */
 sock_handle_t socket_handle_invalid();
 
+/*
+ * Join a multicast address for all UDP sockets. This will allow reception of multicast packets
+ * sent to the given address on all UDP sockets which have bound the port to which the multicast
+ * packet was sent.
+ * @param addr IP multicast address to join
+ * @return Return the result of the join operation
+ */
+sock_result_t socket_join_multicast(const sockaddr_t *addr);
```

Comment:
Multicast makes sense on IPv6. To be honest, I could not find the red line telling me where to use the HAL_IPAddress and where to use sockaddr_t, but it should be easy to change it in that direction.

The network_interface_t makes sense, I will add it. Not connected for the moment, right? And nice to hear about the multiple interfaces :-).

Sorry, I don't really get the reason for the reserved parameter. Why would this be necessary? Is this done in other functions as well?